src/Parser/VengeanceDemonHunter/CombatLogParser.js

```diff
@@ -89,23 +91,28 @@ class CombatLogParser extends MainCombatLogParser {
   }
 
   generateResults() {
+      
     const results = super.generateResults();
 
     const fightDuration = this.fightDuration;
 
-    const nonDpsTimePercentage = this.modules.alwaysBeCasting.totalDamagingTimeWasted / fightDuration;
     const deadTimePercentage = this.modules.alwaysBeCasting.totalTimeWasted / fightDuration;
 
-    console.log(this.modules.alwaysBeCasting.totalTimeWasted+'---'+fightDuration);
+    // As soon as the information is ready to be analysed, gets it and put it in variables
+    // This is done to not get the 'cannot read property of undefined' error
+    if(this.modules.abilityTracker.abilities[SPELLS.SOUL_FRAGMENT.id] !== undefined)  {
```

Comment:
I've been looking to this and I thought it would be better to not removed it. I don't know if you agree with me, but sometimes redundancy and extra code makes a better understanding of the overall project. In that case, I'm letting explicitly that I just want to wait until one of the spells go defined. So can I let it there, as it doesn't change anything to the logic?